pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -433,3 +433,70 @@ func (n *NodeLabeller) loadKSM() {
 	n.KSM.Enabled = bytes.Equal(ksmValue, []byte("1\n"))
 	return
 }
+
+// shouldEnableKSM will update the ksm of the node (if available) based on the kv configuration.
+// If the node labels match the selector terms, the ksm will be enabled.
+// Empty Selector will enable ksm for every node
+func (n *NodeLabeller) shouldEnableKSM(node *v1.Node) {
+	ksmConfig := n.clusterConfig.GetKSMConfiguration()
+	if ksmConfig == nil {
+		n.disableKSM(node)
+		return
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(ksmConfig.NodeLabelSelector)
+	if err != nil {
+		log.DefaultLogger().Errorf("An error occurred while converting the ksm selector: %s", err)
+		return
+	}
+
+	if !selector.Matches(labels.Set(node.ObjectMeta.Labels)) {
+		n.disableKSM(node)
+		return
+	}
+
+	n.enableKSM(node)
+}
+
+func (n *NodeLabeller) enableKSM(node *v1.Node) {
+	currentKSMValue, err := os.ReadFile(n.KSM.SysfsFilePath)
```

Comment:
Can't we run `loadKSM()` before `shouldEnableKSM()` and just check `n.KSM.Enabled` here?

Reply:
Great catch! Thanks! But for this I also need to introduce `Available` field. I also arranged the function name. PTAL